From 95ff340a74af16cca89fd5c5ca99890821053209 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 22 Apr 2017 00:47:16 -0700
Subject: [PATCH] ARM code has unreachable code after switch statement move
 initialization

Fixed
sysdeps/linux-gnu/arm/trace.c:173:33: error: statement will never be executed [-Werror=switch-unreachable]
    uint32_t operand1, operand2, result = 0;
                                 ^~~~~~

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
Upstream-Status: Pending

 sysdeps/linux-gnu/arm/trace.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sysdeps/linux-gnu/arm/trace.c b/sysdeps/linux-gnu/arm/trace.c
index 5e51e91..f974d1f 100644
--- a/sysdeps/linux-gnu/arm/trace.c
+++ b/sysdeps/linux-gnu/arm/trace.c
@@ -155,6 +155,8 @@ arm_get_next_pcs(struct process *proc,
 	const unsigned cond = BITS(this_instr, 28, 31);
 	const unsigned opcode = BITS(this_instr, 24, 27);
 
+	uint32_t operand1, operand2, result = 0;
+
 	if (cond == COND_NV)
 		switch (opcode) {
 			arch_addr_t addr;
@@ -170,7 +172,6 @@ arm_get_next_pcs(struct process *proc,
 		}
 	else
 		switch (opcode) {
-			uint32_t operand1, operand2, result = 0;
 		case 0x0:
 		case 0x1:			/* data processing */
 		case 0x2:
-- 
2.12.2

